presto-raptorx/src/main/java/com/facebook/presto/raptorx/metadata/CommitCleaner.java

```diff
@@ -175,6 +195,41 @@ public synchronized void removeOldCommits()
         }
     }
 
+    // TODO(taozhao) For workers, shardDao doesn't have to be a list, each one only deals with its own node_id. May write a separate Cleaner later.
+    public void removeOldWorkerTransactions(long nodeId)
+    {
+        ShardCommitCleanerDao thisNodeDao = shardDao.get(dbShard(nodeId, shardDao.size()));
+
+        Set<Long> transactionIds = thisNodeDao.getSuccessfulTransactionIds(nodeId);
+        // May need to shard CreatedChunks by nodeId too; but in the near future there is only one Mysql, so can live with it for some time.
```

Comment:
Make it a TODO

Reply:
I remove some "TODO" in this file. Turns out, createdChunks and transactions and table_sizes need to be in the same db. 

So they all shard by TableID for now.